### DISCUSSION TITLE
boundary conditions in heat conduction

### DISCUSSION THREAD

**Discussion by ali-mhassaan** [idaholab/moose#17504] [Q&A Modules: General] [answered]
Hi, 
I'm using the heat conduction module to apply a source term on my system, my questions are:
1. Which MOOSE boundary condition should I use to apply combined convective and radiative bcs? 
2. Also, how can I apply the boundary condition on the surface where I'm applying the source term, after the source has been removed?

Thanks,
Hassaan

**Comment by cticenhour** [comment, marked as answer]
Because convective and radiative BCs in the heat conduction module are Integrated BCs, they have an additive effect when used together in an input file. So, for example, adding [ADConvectiveHeatFluxBC](https://mooseframework.inl.gov/source/bcs/ADConvectiveHeatFluxBC.html) and [FunctionRadiativeBC](https://mooseframework.inl.gov/source/bcs/FunctionRadiativeBC.html) to the same boundary would mean they both take effect in your simulation result. 

As far as question (2), you may want to look into the [TimePeriod control object](https://mooseframework.inl.gov/source/controls/TimePeriod.html) as it allows control of when objects in the input file are active or inactive and for what timeframe in the simulation. You can find more information about the MOOSE Controls system in general at [this link](https://mooseframework.inl.gov/syntax/Controls/index.html).

**Reply by ali-mhassaan** [reply]
Thank you Casey for your response.

Since I kept getting errors when I applied two boundary conditions on the same boundary, I just edited an existing boundary condition. 
Here's what a part of the source file looks like: 

ADRadConHeatFluxBC::ADRadConHeatFluxBC(const InputParameters & parameters)
  : ADIntegratedBC(parameters),
    _T_infinity(getADMaterialProperty<Real>("T_infinity")),
    _htc(getADMaterialProperty<Real>("convective_heat_transfer_coefficient")),
    _rhtc(getADMaterialProperty<Real>("radiation_heat_transfer_coefficient"))
{
}

ADReal
ADRadConHeatFluxBC::computeQpResidual()
{
  return -_test[_i][_qp] * ((_htc[_qp] * (_T_infinity[_qp] - _u[_qp])) + (_rhtc[_qp] * ((_T_infinity[_qp] * _T_infinity[_qp] * _T_infinity[_qp] * _T_infinity[_qp]) - (_u[_qp] * _u[_qp] * _u[_qp] * _u[_qp]))));
}

Here's what it looks like in the input file:
  [./left]
    type = ADRadConHeatFluxBC
    variable = temp
    boundary = 4
    T_infinity = 293.0
    convective_heat_transfer_coefficient = 2.0e-5
    radiation_heat_transfer_coefficient = 2.84e-14
  [../]

Also, I'll look at the TimePeriod control object and post it here if I need more help.

Cheers,
Hassaan.

**Reply by cticenhour** [reply]
I'm glad you were able to code something up that worked for you, but I am curious about the issue you had. Could you post the error you received when using the two BCs at once? Also which BCs were you using?

**Reply by ali-mhassaan** [reply]
I'll try to recreate the error and post it here.
Can you provide an example in which two boundary conditions have been applied to the same boundary?
Is this how you do it?
[./right]
  type = FunctionRadiativeBC
  variable = temp
  boundary = 'right'
  emissivity_function = '3/(5.670367e-8 * 4 * 300 * 300 * 300)'
[../]
[./right]
  type = ADConvectiveHeatFluxBC
  variable = temp
  boundary = 'right'
  T_infinity = 100.0
  heat_transfer_coefficient = 1
[../]

**Reply by cticenhour** [reply]
You cannot have two BCs with the same name (`./right` in your example), so that would cause an error at runtime (a complaint about parameters being supplied multiple times). Change one or both of them to another name. Otherwise, the parameters for your objects look fine to me.
